### PR TITLE
MAISTRA-788 update operator-sdk to v0.10

### PR DESCRIPTION
**Pull request by rcernich**
The vendor updates are in a separate commit for easier reviewing.

### CODE REVIEW

File: pkg/controller/common/util.go [hunk 1]
```diff
@@ -114,24 +111,13 @@ func SetAnnotation(resource metav1.Object, annotation, value string) {
 var initOperatorNamespace sync.Once
 var operatorNamespace string
 
-// XXX: remove after updating to newer version of operator-sdk
-// from newer version of operator-sdk
+// GetOperatorNamespace initializes and caches this operator's namespace; panics on error
 func GetOperatorNamespace() string {
 	initOperatorNamespace.Do(func() {
-		if ns := os.Getenv("POD_NAMESPACE"); ns != "" {
```
Comment (dgn) [inline pkg/controller/common/util.go:121]: Maybe we should leave this in; we added this to run the operator locally as if it was deployed to `POD_NAMESPACE` without having to write to `/var/run/secrets/kubernetes.io/serviceaccount/namespace` locally.

Comment (rcernich) [reply pkg/controller/common/util.go:121]: Added it back in, if `ErrNoNamespace` is returned.

File: tools.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// +build tools
```
Comment (dgn) [inline tools.go:1]: what is this used for?

File: cmd/manager/main.go [hunk 3]
```diff
@@ -99,25 +115,63 @@ func main() {
...
+// It serves those metrics on "http://metricsHost:operatorMetricsPort".
+func serveCRMetrics(cfg *rest.Config) error {
+	// Below function returns filtered operator/CustomResource specific GVKs.
+	// For more control override the below GVK list with your own custom logic.
+	filteredGVK, err := k8sutil.GetGVKsFromAddToScheme(maistrav1.SchemeBuilder.AddToScheme)
+	if err != nil {
+		return err
+	}
+	// Get the namespace the operator is currently deployed in.
+	operatorNs, err := k8sutil.GetOperatorNamespace()
```
Comment (dgn) [inline cmd/manager/main.go]: I think we should use our `util.GetOperatorNamespace()` here for consistency

**Review by luksa — APPROVED** [review]

**Review by dgn — APPROVED** [review]
looks good, just one (very) minor comment and a question

ah btw we have to update `DEVEL.md` to remove `--discoveryCacheDir`

File: cmd/manager/main.go (hunk 3 above)
Comment (rcernich) [reply cmd/manager/main.go]: good catch, i'll update that.

File: tools.go (hunk 2 above)
Comment (rcernich) [reply tools.go:1]: I'm not 100% certain, but I believe this is required to use the k8s.io/code-generator tooling with go modules.  The file is generated by operator-sdk.

**Comment by rcernich** [comment]
@luksa, let me know if the controller-runtime changes should be in here or not.  (I changed this PR back to WIP.)

**Pull request merged**